### PR TITLE
Bluetooth: Mesh: Remove outdated RPL entries in settings work item

### PR DESCRIPTION
**Pull request by PavelVPV**
This allows to move flash operations from the context that triggered IVI update to the mesh settings work item that runs on the system workqueue. After this change, the outdated RPL entries will be removed in the setting work item triggered by store_pending function in mesh/settings.c. This is required to for a case where the mesh settings work item is running on a separate thread instead of the system workqueue to unblock the system workqueue eventually.

### CODE REVIEW

File: subsys/bluetooth/mesh/rpl.c [hunk 1]
```diff
@@ -118,34 +124,33 @@ bool bt_mesh_rpl_check(struct bt_mesh_net_rx *rx,
...
-			return false;
+			goto match;
 		}
 
 		/* Existing slot for given address */
 		if (rpl->src == rx->ctx.addr) {
+			if (!rpl->old_iv &&
+			    atomic_test_bit(&rpl_flags, PENDING_RESET) &&
+			    !atomic_test_bit(store, i)) {
+				/* Until rpl reset is finished, entry with old_iv == false and
```
Comment (alxelax) [inline subsys/bluetooth/mesh/rpl.c:139]: Could you clarify it a bit more? Why does it happen for `old_iv == false` and doesn't for `old_iv == true`?

Comment (alxelax) [reply subsys/bluetooth/mesh/rpl.c:139]: `store` is the map of `old_iv` flags. If `old_iv` is false then `store` is set. Does it have sense to check both `!rpl->old_iv` and `!atomic_test_bit(store, i)`?

**Review by alxelax — COMMENTED** [review]
LGTM but I'd like to get clarification for my questions.

**Review by alxelax — DISMISSED** [review]
btw. scenario with removing rpl entries and receiving some frame\frames during the reset procedure is quite complex. It would be nice to have bsim test for it.

**Comment by PavelVPV** [comment]
> btw. scenario with removing rpl entries and receiving some frame\frames during the reset procedure is quite complex. It would be nice to have bsim test for it.

Not sure how to test it in bsim as a message needs to be sent in a specific moment. Unit testing should be possible.

**Review by alxelax — DISMISSED** [review]
Very good tests btw. Now there are acceptance criteria for any RPL related changes.

File: tests/bluetooth/mesh/rpl/src/main.c [hunk 2]
```diff
@@ -0,0 +1,480 @@
...
+{
+	prepare_rpl_and_start_reset();
+
+	for (int i = 0; i < ARRAY_SIZE(test_vector); i++) {
+		if (test_vector[i].old_iv) {
+			ztest_expect_value(settings_delete, name, test_vector[i].name);
+		} else {
+			ztest_expect_data(settings_save_one, name, test_vector[i].name);
+		}
+	}
```
Comment (Andrewpini) [inline tests/bluetooth/mesh/rpl/src/main.c:252]: Nitpick: Could you use `expect_pending_store()` here? Seems to be identical.

File: tests/bluetooth/mesh/rpl/src/main.c [hunk 3]
```diff
@@ -0,0 +1,480 @@
...
+
+	bt_mesh_rpl_pending_store(BT_MESH_ADDR_ALL_NODES);
+	zassert_true(is_rpl_check_called());
+
+	/* Call bt_mesh_rpl_pending_store() to store new entry. */
+	ztest_expect_data(settings_save_one, name, entry->name);
+	bt_mesh_rpl_pending_store(BT_MESH_ADDR_ALL_NODES);
+
+	check_entries_from_test_vector();
+	check_empty_entries(EMPTY_ENTRIES_CNT);
```
Comment (Andrewpini) [inline tests/bluetooth/mesh/rpl/src/main.c:285]: Nitpick: Repeated pattern for several of the tests. Could it be bundled in a separate func?

File: tests/bluetooth/mesh/rpl/src/main.c [hunk 4]
```diff
@@ -0,0 +1,480 @@
...
+	prepare_rpl_and_start_reset();
+	expect_pending_store();
+
+	/* Take the first entry with old_iv == false and simulate msg reception with same src
+	 * address and correct IVI after the entry was stored.
+	 */
+	struct test_rpl_entry *entry = &test_vector[0];
+
+	zassert_false(entry->old_iv);
+	entry->old_iv = false;
```
Comment (Andrewpini) [inline tests/bluetooth/mesh/rpl/src/main.c:302]:
```suggestion
	zassert_false(entry->old_iv);
```
Is it not redundant to set the old_iv to false after checking that it is already set to false?

**Review by Andrewpini — COMMENTED** [review]
As far as I can see this looks good. Appreciate the unit test 👍 I want to take a little more time to go through the test cases, but this is my feedback so far.

**Review by alxelax — APPROVED** [review]

**Review by Andrewpini — APPROVED** [review]

**Pull request merged**